centreon-certified/prometheus/prometheus-pushgateway-metrics-apiv2.lua

```diff
@@ -0,0 +1,613 @@
+-- ifnumber_not_nan: [for Prometheus] check if a number is a number (and not a NaN)
+-- @param {number} number, the number to check
+-- @return {boolean}
+--------------------------------------------------------------------------------
+local function ifnumber_not_nan (number)
+  if (number ~= number) then
+    return false
+  elseif (type(number) ~= 'number') then
+    return false
+  else
+    return true
+  end
+end
+
+--------------------------------------------------------------------------------
+-- get_metric_type: [for Prometheus] find out the metric type to match openmetrics standard
+-- @param {table} perfdata, the perfdata informations
+-- @return {string} metricType, the type of the metric
+--------------------------------------------------------------------------------
+function EventQueue:get_metric_type (perfdata)
+  local metricType = nil;
+  if (ifnumber_not_nan(perfdata.max)) then
+    metricType = 'gauge'
+  else 
+    metricType = 'counter'
+  end
+  
+  return metricType
+end
+
+--------------------------------------------------------------------------------
+-- EventQueue:add, add an event to the sending queue
+--------------------------------------------------------------------------------
+function EventQueue:add()
+  -- store event in self.events lists
+  local category = self.sc_event.event.category
+  local element = self.sc_event.event.element
+
+  self.sc_logger:debug("[EventQueue:add]: add event in queue category: " .. tostring(self.sc_params.params.reverse_category_mapping[category])
+    .. " element: " .. tostring(self.sc_params.params.reverse_element_mapping[category][element]))
+
+  self.sc_logger:debug("[EventQueue:add]: queue size before adding event: " .. tostring(#self.sc_flush.queues[category][element].events))
+  self.sc_flush.queues[category][element].events[#self.sc_flush.queues[category][element].events + 1] = self.sc_event.event.formated_event
+
+  self.sc_logger:info("[EventQueue:add]: queue size is now: " .. tostring(#self.sc_flush.queues[category][element].events) 
+    .. ", max is: " .. tostring(self.sc_params.params.max_buffer_size))
+end
+
+--------------------------------------------------------------------------------
+-- EventQueue:build_payload, concatenate data so it is ready to be sent
+-- @param payload {string} json encoded string
+-- @param event {table} the event that is going to be added to the payload
+-- @return payload {string} json encoded string
+--------------------------------------------------------------------------------
+function EventQueue:build_payload(payload, event)
+
+  if not payload then -- FIXME: voir obsidian
+    payload = event
+  else
+    table.insert(payload, event)
+  end
+
+  return payload
+end
+
+function EventQueue:send_data(payload, queue_metadata)
+  self.sc_logger:debug("[EventQueue:send_data]: Starting to send data")
+  local httpPostData = payload.payload
+  local httpResponseBody = ""
+
```

Comment:
missing the send_data_test and log_curl_command feature

Reply:
fixed here https://github.com/centreon/centreon-stream-connector-scripts/pull/279/commits/b386ac65b9d1c10786fe7a9be34ca8fb570f334e